### ISSUE TITLE
Support rust filters in the Envoy proxy of consul

### ISSUE THREAD

**Issue by kanishkarj** [nicholasjackson/envoy-experiments#2]
I tried loading a WASM filter written in rust onto consul. Currently, it throws this error: "WASM missing malloc/free". A little digging lead me to this github issue : https://github.com/proxy-wasm/proxy-wasm-rust-sdk/issues/3 . So I guess having consul use the "istio/proxyv2:1.5.0" image of envoy would help support filters in Rust (the ones using the proxy-wasm-rust-sdk) and maybe other languages. Is it possible to generate a consul image that uses this image of envoy?

Thank you

**Comment by nicholasjackson** [comment]
100%, I will do that today.

**Comment by kanishkarj** [comment]
Great! :)

**Comment by nicholasjackson** [comment]
Hey @kanishkarj I have pushed the updates to this repo which allow you to use the latest Envoy with WASM as used by Istio.

I have also built a combo image which contains the Consul dev version and also the Envoy version

nicholasjackson/consul-envoy:dev-dev

If you wanted to use these in the Helm chart I just had a thought, at present there is no way to specify the WASM filter which will be loaded by the sidecar. Just going to do a dev build of our controller and will add an example to this repo today.
